substratevm/src/com.oracle.objectfile/src/com/oracle/objectfile/debugentry/ClassEntry.java

```diff
@@ -313,8 +364,99 @@ public int lowpc() {
      *
      * @return the highest code section offset for compiled method code belonging to this class
      */
+    @SuppressWarnings("unused")
     public int hipc() {
         assert hasCompiledEntries();
         return compiledEntries.get(compiledEntries.size() - 1).getPrimary().getHi();
     }
+
+    /**
+     * Add a file to the list of files referenced from info associated with this class.
+     * 
+     * @param fileEntry The file to be added.
+     */
+    public void includeFile(FileEntry fileEntry) {
+        if (files == null) {
+            files = new ArrayList<>();
+        }
+        assert !files.contains(fileEntry);
+        // cannot add files after index has been created
+        assert fileIndex == null;
+        files.add(fileEntry);
+    }
+
+    /**
+     * Add a directory to the list of firectories referenced from info associated with this class.
+     * 
+     * @param dirEntry The directory to be added.
+     */
+    public void includeDir(DirEntry dirEntry) {
+        if (dirs == null) {
+            dirs = new ArrayList<>();
+        }
+        assert !dirs.contains(dirEntry);
+        // cannot add dirs after index has been created
```

Comment:
should this be the assertion description?

Reply:
I have updated the asserts to include the relevant comment as the assert message.